WordPress/src/test/java/org/wordpress/android/ui/stats/refresh/StatsViewModelTest.kt

```diff
@@ -240,11 +240,12 @@ class StatsViewModelTest : BaseUnitTest() {
     @Test
     fun `given enable stats module is clicked, then state reflects progress`() = test {
         val uiModel = initObservers().uiModelObserver
+        whenever(statsModuleActivateUseCase.postActivateStatsModule(anyOrNull())).thenReturn(success)
 
         startViewModel(statsModuleEnabled = false)
         viewModel.onEnableStatsModuleClick()
 
-        assertThat(uiModel.last().disabledStatsProgressVisible).isTrue
+        assertThat(uiModel.last().disabledStatsProgressVisible).isFalse()
     }
```

Comment:
The test after change tests something different than initially. I think we shouldn't change te test body, but instead fix the issue that breaks this test which is `kotlin.NoWhenBranchMatchedException` in `StatsViewModel#onEnableStatsModuleClick` method.

Reply:
Yes, this is exactly my thought process @wzieba , and I did briefly tried to make it (somehow) work, timeboxing my efforts, but in this specific situation, the test was wrong, the `last` value of `disabledStatsProgressVisible` ends up being `false` by the end of the test. As such, the `true` value is only set momentarily. This test was only working as is because the `when` branch wasn't working as expected. Does that make sense? 🤔

Reply:
> As such, the true value is only set momentarily

That's true, and I think this is what the test wanted to test: that after `onEnableStatsModuleClick` is tapped, and `postActivateStatsModule` method is not finished, we should show "disabled stats progress". 

If you add an empty `else` to `onEnableStatsModuleClick` to fix `NoWhenBranchMatchedException` and revert changes to this test made in this PR, the test will pass. 

I prefer we do this, because `NoWhenBranchMatchedException` is a runtime exception, so we would start throwing it in production app if not fixed.

Reply:
Ah I see, mockito returns null for `postActivateStatsModule` and this seems to trigger `NoWhenBranchMatchedException`, so not really a production-possible scenario

Reply:
😃 That's there is the problem @wzieba , adding an `else` there is wrong, it being redundant and all, this `when` is already covering all cases of `StatsModuleActivateRequestState`, which are:
- Success
- Failure.NetworkUnavailable
- Failure.RemoteRequestFailure

The problem with the test was that they didn't specify the `whenever(...).thenReturn(...)` test setup (`given`) to address the path, thus the test was failing on `NoWhenBranchMatchedException`. Now, no matter the path you chose, in the end the `disabledStatsProgressVisible` end up being `false`.

Does that help clarify? 🙏

Reply:
Either way, to not alter test logic and test that `disabledStatsProgressVisible` is indeed `true` without going to `when` (which is now problematic) I'd suggest using the `StandardTestDispatcher` instead:

```kotlin
   @Test
    fun `given enable stats module is clicked, then state reflects progress`() = runTest(StandardTestDispatcher()) {
        val uiModel = initObservers().uiModelObserver

        startViewModel(statsModuleEnabled = false)
        viewModel.onEnableStatsModuleClick()

        assertThat(uiModel.last().disabledStatsProgressVisible).isTrue()
    }
```

Reply:
Why do you think this solves the problem? 🤔

1. I tried it just now and I still get the `kotlin.NoWhenBranchMatchedException` thrown.
2. IMHO we are having a wrong setup, not showing a clear path for the `when` case to progress further, thus an incomplete test which are hard to reason about.

Reply:
No other ideas from me, this should be probably written differently to have a robust test for checking if "disabled state progress" has been shown. Thanks for the discussion, closing as resolved!

Reply:
Exactly my thinking, thus, I first considered removing it altogether, but then I chose to update it. Maybe we should actually remove it, just to keep things simple, wdyt @wzieba ? 🤔